contracts/RFQ.sol

```diff
@@ -144,19 +145,59 @@ contract RFQ is IRFQ, StrategyBase, ReentrancyGuard, SignatureValidator, BaseLib
         // Transfer token/Eth to receiver
         if (_order.makerAssetAddr == address(weth)) {
             // Transfer from maker
-            spender.spendFromUser(_order.makerAddr, _order.makerAssetAddr, settleAmount);
+            _spendFromMakerTo(_order.makerAddr, _order.makerAssetAddr, address(this), settleAmount, _useSpenderForMaker);
             weth.withdraw(settleAmount);
             payable(_order.receiverAddr).transfer(settleAmount);
         } else {
-            spender.spendFromUserTo(_order.makerAddr, _order.makerAssetAddr, _order.receiverAddr, settleAmount);
+            _spendFromMakerTo(_order.makerAddr, _order.makerAssetAddr, _order.receiverAddr, settleAmount, _useSpenderForMaker);
         }
         // Collect fee
         if (fee > 0) {
-            spender.spendFromUserTo(_order.makerAddr, _order.makerAssetAddr, feeCollector, fee);
+            _spendFromMakerTo(_order.makerAddr, _order.makerAssetAddr, feeCollector, fee, _useSpenderForMaker);
         }
 
         _emitFillOrder(_order, _vars, settleAmount);
 
         return settleAmount;
     }
+
+    function _spendFromMakerTo(
+        address _user,
+        address _tokenAddr,
+        address _recipient,
+        uint256 _amount,
+        bool _useSpenderForMaker
+    ) internal {
+        if (_useSpenderForMaker) {
+            spender.spendFromUserTo(_user, _tokenAddr, _recipient, _amount);
+            return;
+        }
+        uint256 balanceBefore = IERC20(_tokenAddr).balanceOf(_recipient);
```

Comment:
We may need to review the needs of this before/after check under different situations.

Reply:
If we don't check balance match, then transfers of inflationary tokens would succeed. Since we now transfer all fee away from strategy contract, allowing inflationary tokens would not hurt us but the numbers would probably not match between expected amount we displayed to user and user's actual received amount.

Reply:
Since maker only provide quotes to certain pair, inflationary tokens can be avoided in RFQ. I think there's no reason for a maker or user to skip our API but it's an assumption.

Reply:
if we abstract this to a common contract, check balance may be better?

Reply:
Then the argument is based on that RFQ must share a common spending function from other contracts. But if we replace this inheritance by one line ERC20 transfering, we can save some gas in RFQ.